initializr-generator-spring/src/main/java/io/spring/initializr/generator/spring/scm/git/GitIgnore.java

```diff
@@ -28,51 +30,37 @@
  * popular IDEs. Empty sections are not rendered.
  *
  * @author Stephane Nicoll
+ * @author Yves Galvão
  */
 public class GitIgnore {
 
-	private final GitIgnoreSection general = new GitIgnoreSection(null);
+	private final Map<GitEnum, GitIgnoreSection> ignoreSections;
 
-	private final GitIgnoreSection sts = new GitIgnoreSection("STS");
```

Comment:
The idea of this is to model the `.gitgnore` files with well-defined sections. If you don't set them, they are not rendered (as the class javadoc states). I don't understand why moving to a map based approach is better. Can you expand on that?

Reply:
It's not the write method on GitIgnore thas is responsible for render the GitIgnoreSection informations on .gitignore file? If not, maybe I have a misinterpretation of the flow, and this PR does not make sense.